Automattic/wp-calypso
# Gravatar: Customize the "Welcome back!" view of the passwordless login flow

Related to 105814-Automattic/gravatar

## Proposed Changes

* To customize the "Welcome back!" view for Gravatar-powered clients

## Testing Instructions



* Sandbox `public-api.wordpress.com`, and then apply this diff: `arc patch D122814`
* Setup Calypso, and run `yarn start`
* Log out of your account, and log in to [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3Dbfa7ea95b0a07d39e3f23e1d923ce77b35016295d986b51361d4334a6e571206%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1). You will see the "Welcome back!" views have been customized as below:

| Grav | WPJM |
| - | - |
| <img width="765" alt="l_g" src="https://github.com/Automattic/wp-calypso/assets/21308003/8003aaf8-a64a-4232-a058-b74394647905"> | <img width="737" alt="l_j" src="https://github.com/Automattic/wp-calypso/assets/21308003/4c219b1a-1386-499b-aaf7-26f8f9c47f02"> |

* Log out of your account. [Disable this submission function](https://github.com/Automattic/wp-calypso/blob/8d6eddd829e42129d7f251b61101174a2a7fc82a/client/login/magic-login/handle-emailed-link-form.jsx#L86), and log in to [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3Dbfa7ea95b0a07d39e3f23e1d923ce77b35016295d986b51361d4334a6e571206%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1). You will see the "Welcome back!" views have been customized as below:

| Grav | WPJM |
| - | - |
| <img width="706" alt="b_g" src="https://github.com/Automattic/wp-calypso/assets/21308003/803e047a-fbf6-4c08-85a1-103a837ea1d4"> | <img width="706" alt="b_j" src="https://github.com/Automattic/wp-calypso/assets/21308003/bbd8bbf3-6445-4876-bdf5-9ca42cf7693a"> |

* Log out of your account. Go to your mailbox and then click an expired Gravatar-powered client's login link. The expired link view should still work the same:

<img width="902" alt="截圖 2023-09-25 下午3 10 58" src="https://github.com/Automattic/wp-calypso/assets/21308003/7ddc0459-bec7-4bf5-b764-d870c49f5dbb">

* The WPCOM's [magic login](http://calypso.localhost:3000/log-in/link) still work the same.

## Pre-merge Checklist
